File: src/src/org/github/olloginov/ideataskbrowser/view/TaskTreeRenderer.java
package org.github.olloginov.ideataskbrowser.view;

import com.intellij.ui.ColoredTreeCellRenderer;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public class TaskTreeRenderer extends ColoredTreeCellRenderer {
    @Override
    public void customizeCellRenderer(@NotNull JTree tree, Object value, boolean selected, boolean expanded, boolean leaf, int row, boolean hasFocus) {
        if (value instanceof CustomIcon) {
            setIcon(((CustomIcon) value).getIcon());
        }
        if (value instanceof CustomLabel) {
            ((CustomLabel) value).setLabel(this);
        }
    }
}

File: src/main/kotlin/org/github/olloginov/ideataskbrowser/view/TaskTreeRenderer.kt
package org.github.olloginov.ideataskbrowser.view

import com.intellij.ui.ColoredTreeCellRenderer
import javax.swing.JTree

class TaskTreeRenderer : ColoredTreeCellRenderer() {
	override fun customizeCellRenderer(tree: JTree, value: Any?, selected: Boolean, expanded: Boolean, leaf: Boolean, row: Int, hasFocus: Boolean) {
		if (value is CustomIcon) {
			setIcon(value.getIcon())
		}
		if (value is CustomLabel) {
			value.setLabel(this)
		}
	}
}
